public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -189,62 +189,61 @@ export class GraphiteDatasource
   }
 
   query(options: DataQueryRequest<GraphiteQuery>): Observable<DataQueryResponse> {
-    const streams: Array<Observable<DataQueryResponse>> = [];
-
-    for (const target of options.targets) {
-      // hiding target is handled in buildGraphiteParams
-      if (target.fromAnnotations) {
-        streams.push(
-          new Observable((subscriber) => {
-            this.annotationEvents(options.range, target)
-              .then((events) => subscriber.next({ data: [toDataFrame(events)] }))
-              .catch((ex) => subscriber.error(new Error(ex)))
-              .finally(() => subscriber.complete());
-          })
-        );
-      } else {
-        // handle the queries here
-        const graphOptions = {
-          from: this.translateTime(options.range.from, false, options.timezone),
-          until: this.translateTime(options.range.to, true, options.timezone),
-          targets: options.targets,
-          format: (options as GraphiteQueryRequest).format,
-          cacheTimeout: options.cacheTimeout || this.cacheTimeout,
-          maxDataPoints: options.maxDataPoints,
-        };
-
-        const params = this.buildGraphiteParams(graphOptions, options.scopedVars);
-        if (params.length === 0) {
-          return of({ data: [] });
-        }
-
-        if (this.isMetricTank) {
-          params.push('meta=true');
+    if (options.targets.some((target: GraphiteQuery) => target.fromAnnotations)) {
+      const streams: Array<Observable<DataQueryResponse>> = [];
+
+      for (const target of options.targets) {
+        // hiding target is handled in buildGraphiteParams
+        if (target.fromAnnotations) {
```

Comment:
I think we do not need this condition anymore since we are already filtering these on line 193

Reply:
Done, removed the duplicate condition. Thank you!